apps/prairielearn/src/schemas/infoCourseInstance.ts

```diff
@@ -58,16 +58,9 @@ export const CourseInstanceJsonSchema = z
         beforeDate: z
```

Comment:
Drive-by comment: I find myself really wishing we had an `allow` prefix on this. The JSON in its current form looks strange:

```json
"selfEnrollment": {
  "useEnrollmentCode": true,
  "beforeDate": "2025-10-07T13:23:58"
}
```

This feels better to me:

```json
"selfEnrollment": {
  "useEnrollmentCode": true,
  "allowBeforeDate": "2025-10-07T13:23:58"
}
```